src/Compilers/Core/Portable/MetadataReference/ModuleMetadata.cs

```diff
@@ -74,6 +85,33 @@ private ModuleMetadata(ModuleMetadata metadata)
         /// <exception cref="ArgumentNullException"><paramref name="metadata"/> is null.</exception>
         /// <exception cref="ArgumentOutOfRangeException"><paramref name="size"/> is not positive.</exception>
         public static ModuleMetadata CreateFromMetadata(IntPtr metadata, int size)
+            => CreateFromMetadataWorker(metadata, size, owner: null, disposeOwner: false);
+
+        /// <summary>
+        /// Create metadata module from a raw memory pointer to metadata directory of a PE image or .cormeta section of an object file.
+        /// Only manifest modules are currently supported.
+        /// </summary>
+        /// <param name="metadata">Pointer to the start of metadata block.</param>
+        /// <param name="size">The size of the metadata block.</param>
+        /// <param name="owner">Data that should be kept alive as long as this <see cref="ModuleMetadata"/> is alive.  This can be
+        /// useful, for example, if there is backing memory that the metadata depends on that should be kept rooted so it
+        /// doesn't get garbage collected.</param>
+        /// <param name="disposeOwner">Whether or not <paramref name="owner"/> should be <see cref="IDisposable.Dispose"/>'d when this object is
+        /// Disposed.</param>
+        public static unsafe ModuleMetadata CreateFromMetadata(
+            IntPtr metadata,
+            int size,
+            IDisposable owner,
+            bool disposeOwner)
+        {
+            return CreateFromMetadataWorker(metadata, size, owner, disposeOwner);
+        }
+
+        private static ModuleMetadata CreateFromMetadataWorker(
```

Comment:
The signatures for `CreateFromMetadata(4 parameters)` and `CreateFromMetadataWorker(4 parameters)` are identical. Can we remove `CreateFromMetadataWorker`?

Reply:
This is no longer relevant assuming we add the suggested null check to public API.